File: javascript/example_code/nodejs/src/connection-util.js
import pg from "pg";
import { generateToken } from "./token-gen.js";
const { Client } = pg;

async function getClient(clusterEndpoint, region) {
    const action = "DbConnectSuperuser";
    let token;
    try {
        // The token expiration time is optional, and the default value 900 seconds
        token = await generateToken(clusterEndpoint, action, region);
        const client = new Client({
            host: clusterEndpoint,
            user: "admin",
            password: token,
            database: "postgres",
            port: 5432,
            ssl: {
                require: true,
                rejectUnauthorized: false
            }
        });
        await client.connect();
        return Promise.resolve(client);
    } catch (error) {
        return Promise.reject(error);
    }
}

export { getClient }

File: dotnet/example_code/npgsql/example/ConnectionUtil.cs
using Npgsql;
using Amazon;

namespace Example
{
    public static class ConnectionUtil
    {
        public static async Task<NpgsqlConnection> GetConnection(string cluster, RegionEndpoint region)
        {
            const string username = "admin";
            // The token expiration time is optional, and the default value 900 seconds
            string password = TokenGenerator.GenerateAuthToken(cluster, region);
            const string database = "postgres";
            var connString = "Host=" + cluster + ";Username=" + username + ";Password=" + password + ";Database=" + database + ";Port=" + 5432 + ";SSLMode=Require;";

            var conn = new NpgsqlConnection(connString);
            await conn.OpenAsync();
            return conn;
        }
    }
}
